cpp/include/raft/comms/std_comms.hpp

```diff
@@ -188,11 +189,87 @@ class std_comms : public comms_iface {
 
   int get_rank() const { return rank_; }
 
+  void scatter_nccluniqueid(ncclUniqueId &id, int color, int key, int root,
```

Comment:
To your first point, `comms_t` contains the public API. Since this is a PIMP, there's no expectation that anyone would be calling this directly. That said, I could be explicit and mark this private if it helps.

Reply:
OK, in that case, full documentation can be an overkill, but some brief explanations about the input parameters can be helpful for any future RAFT developers who need to update this function. It took me a decent amount of time to figure out what these parameters are.